xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererMediaCodecSurface.cpp

```diff
@@ -212,13 +217,14 @@ void CRendererMediaCodecSurface::ReorderDrawPoints()
     dstRect.x1, dstRect.y1, dstRect.Width(), dstRect.Height(),
     adjRect.x1, adjRect.y1, adjRect.Width(), adjRect.Height());
 
+  // Assumption: SurfaceFlinger starts PTS processing on arrival of first frame.
+  // To have the best match between kodi / android clock we start on first RenderUpdate call.
+  // Beside this the first frame should not be rendered before the SurfaceView is resized.
```

Comment:
Why not putting the "ReleaseOutputBuffer(true)" in RenderUpdate, rather than FlipPage, as it was before?
FlipPage is a GUI layer thing, so you are mixing the 2 loops, now.
And, obviously, putting the ReleaseOutBuffer in RenderUpdate assures you the first frame is only showed on RenderUpdate, without flags.